### DISCUSSION TITLE
Advanced Menu Planning

### DISCUSSION THREAD

**Discussion by hluedeke** [Pattern-Projector/pattern-projector#136] [Ideas]
I'd like to kick off the organization of menus, as it's something I've started working on today.

Here's my initial thoughts (and only applies to projection at the moment):

Main/Basic Menu:
Flip
Rotate
Recenter
Grid Display
Line thickness (?)
Stitch menu

Advanced Menu:
Measuring Lines
Crosshair for flip

**Comment by courtneypattison** [comment]
I agree with that organization!

Another way to think about it is maybe quick versus long actions or one time versus many actions.

- With the new setup, I'm not sure how often recenter will be used, especially since it's the default.
- I'm guessing line thickness and stitch menu are only used once per cutting session.
- Definitely agree with the items in the advanced menu. I'm not sure if anyone other than me will want the measure tool!

Thoughts on where to put the advanced menu?

**Comment by roel-v** [comment]
So I've been playing a bit with how everything in the UI could be organized. I don't mean these as a 'here use this', just as working ideas to explore approaches. The first set of mockups is essentially what is there now, just organized a bit more consistently, and with the layers and stitching menus worked into the same structure as the main toolbar, in a two-step hierarchical menu. The main disadvantage here is that the layers in the layers aren't in a top to bottom list any more.

One thing to note about both approaches is that I don't think there is a need for splitting 'basic' from 'advanced' features (yet), and if there ever turn out to be one, that can be done easily by adding another top-level menu - wouldn't even have to be called 'advanced', could just be split based on functional differences. This would remove the need for a toggle that hides part of the display, which imo is a great barrier for discoverability.

![Home double welted menu](https://github.com/Pattern-Projector/pattern-projector/assets/568567/733c5bc9-8700-4411-8d0f-fbba82b9d521)

![Layers double welted menu](https://github.com/Pattern-Projector/pattern-projector/assets/568567/ccd6b99b-a923-41d4-bbf1-61bf73f6b78f)

![Stitch double welted menu](https://github.com/Pattern-Projector/pattern-projector/assets/568567/0b5fc840-6e74-4a79-8bac-a65b8da45ccc)

This second concept organizes everything in 'bubbles' and 'groups of bubbles'. This makes all UI elements as wide as they are tall, which makes it possible to orient them both on the sides or top/bottom. This way the user can put the menu where it's out of their way, depending on the h/w ratio of their device's display and the layout of the pattern. I think this way it would even become possible to use on a phone, as those tend to be quite tall nowadays, so by putting them in landscape mode and then putting in some sort of scrolling mechanism for the bubbles you could have a quite usable toolbar even with room for only a few of the buttons. This would need further exploration, I may be wrong on this.

![Home circles](https://github.com/Pattern-Projector/pattern-projector/assets/568567/812ca6e1-8d37-457a-adfa-5f553f6ac036)

![Layers circles](https://github.com/Pattern-Projector/pattern-projector/assets/568567/53808104-1a73-4317-ab21-98f433ea2fc0)

![Stitch circles](https://github.com/Pattern-Projector/pattern-projector/assets/568567/b8a08c51-faec-4bba-a1f5-d298bff9a2f8)
